### ISSUE TITLE
Overpressure damage area needs bug fixes

### ISSUE THREAD

**Issue by battleshipfree** [acemod/ACE3#7805]
**Mods (complete and add to the following information):**
- **Arma 3: 1.98 Stable
- **CBA: 3.15 Stable
- **ACE3: 3.13.3 Stable
_Make sure to reproduce the issue with only CBA and ACE3 on a newly created mission!_

**Description:**
RPGs & cannons/overpressure/damage area

**Steps to reproduce:**
1. Get into VR arsenal and test the RPGs: 
Stand in the same row beside an AI, take a small step forward, and make sure backblast area is clear;
Fire the RPG. 
2. Get into VR arsenal and test the tank cannon: 
Hop in a T-80U;
Fire the cannon.

**Expected behavior:**
1. AIs within 5 metres to your side got wounded by overpressure. Seems that the backblast starts from your shoulder. 
2. Overpressure wounds AIs left & right 90 degrees from the breech, not the muzzle. (which is how it supposed to be?)

**Where did the issue occur?**
Everywhere - Dedicated / Self-Hosted Multiplayer / Singleplayer / Editor (Singleplayer) / Editor (Multiplayer) / Virtual Arsenal

**Log Files:**
The issue didn't cause me to crash

**Additional context:**
As far as I know about ace files, I think this issue might come from -
ace_overpressure.pbo/functions/fnc_overpressureDamage.sqf
where overpressure blast should start from 1. the end of the RPG tube & 2. gun muzzle rather than the projectile ("Pos ASL of the projectile" in line 9). Starting from the projectile means that overpressure blast starts somewhere 1. right in front of your shoulder (for RPGs) & 2. around the gun breech (for cannons), and results in a massive injury of friendly units right beside you (an area almost 170 degrees behind the RPG shooter, definitely unrealistic).

**Comment by BaerMitUmlaut** [comment]
Regarding the tanks: Overpressure occurs all around a firing tank.

![grafik](https://user-images.githubusercontent.com/9693842/88558779-4bf97700-d02c-11ea-819c-398703e4210e.png)

**Comment by commy2** [comment]
>Log Files:
>The issue didn't cause me to crash

The log files are not there to debug a crash.

**Comment by battleshipfree** [comment]
Well, what I'm reporting here is basically about the overpressure damage area too large for RPGs and cannons, which means that whoever steps into it may get significantly injured. And this "large area" I'm referring to is kind of left & right 90 degrees from the launch tube / gun breech (here's the point I'm confused about), thus standing right beside the AT weapon / tank may cause injuries.

**Comment by BaerMitUmlaut** [comment]
I don't get what you're confused about, I thought my response was quite clear.

**Comment by battleshipfree** [comment]
Well, sry for that. 
I'm trying to figure out what's in "ace_overpressure.pbo/functions/fnc_overpressureDamage.sqf", and there might be a error in how the overpressure damage area is calculated. 
Your reference should make sense for tank cannons, but for RPGs, this damage area is probably calculated from the projectile, which makes anyone standing right beside the RPG vulnurable to overpressure damage, like what I reported above - "AIs within 5 metres to your side got wounded by overpressure. Seems that the backblast starts from your shoulder".

**Comment by battleshipfree** [comment]
![图片](https://user-images.githubusercontent.com/53341082/89114293-60ff5b80-d4ad-11ea-9b6d-cd3c86e38731.png)

**Comment by battleshipfree** [comment]
This is a little bit tricky - backblast area of FGM-148 Javelin.
![image](https://user-images.githubusercontent.com/53341082/89114402-6f9a4280-d4ae-11ea-870e-83d1651ccc20.png)

**Comment by battleshipfree** [comment]
I'm gonna find some more references, but my consideration is current version of ACE3 makes it hard for dense deployment of fire teams with light AT weapon, i.e. RPG-7 & AT-4 & M72.

**Comment by commy2** [comment]
Still needs RPT file.

**Comment by Whigital** [comment]
How about adding another weapon property, like `GVAR(offset)`, with a value corresponding to the muzzle where the over pressure area is formed.

For an M1A2 (Rheinmetall Rh-120 L/44), this would be `5.28`. Then modify `ace_overpressure_fnc_firedEHOP` to use that offset to change the position before firing the `ace_overpressure` event for nearby entities. That way the over pressure would originate at the muzzle instead of the breech, as pointed out above. Same could be done for launchers, but with negative values.

I can't find anything regarding the over pressure position being offset, only it originating from the projectile position.

**Comment by BaerMitUmlaut** [comment]
But what exactly is the problem about tanks? Overpressure occurs all around the tank, as it should?

**Comment by Whigital** [comment]
Don't get me wrong. I'm neither confirming or denying any bugs, just spitballing ideas for further enhancements.

**Comment by battleshipfree** [comment]
Not quite sure what a RPT file is, sry.
Hope this drawings can make my bug report clear. (Orange areas refer to approximate ace_overpressure damage area, tested in game).
![image](https://user-images.githubusercontent.com/53341082/89856451-e41a6300-dbcb-11ea-8c74-8bb042c2078a.png)
![image](https://user-images.githubusercontent.com/53341082/89856467-ed0b3480-dbcb-11ea-846b-603b0f8897e0.png)
![image](https://user-images.githubusercontent.com/53341082/89856474-f399ac00-dbcb-11ea-8f8f-06412c8232c0.png)
So my point is - 
1. For RPGs, lateral damage areas are too wide (might caused by the calculation originating from the projectile position);
2. For tanks, the shadowed orange area for overpressure damage should mostly be "cautious area" rather than lethal enough to stun friendly troops, while overpressure area should neither be originated from the projectile position, nor a semi-circle shape.

**Comment by battleshipfree** [comment]
> How about adding another weapon property, like `GVAR(offset)`, with a value corresponding to the muzzle where the over pressure area is formed.
> 
> For an M1A2 (Rheinmetall Rh-120 L/44), this would be `5.28`. Then modify `ace_overpressure_fnc_firedEHOP` to use that offset to change the position before firing the `ace_overpressure` event for nearby entities. That way the over pressure would originate at the muzzle instead of the breech, as pointed out above. Same could be done for launchers, but with negative values.
> 
> I can't find anything regarding the over pressure position being offset, only it originating from the projectile position.

I agree with this new variable to determine the point where overpressure effects start from - cannon muzzles & the end of the RPG tube.

**Comment by BlackAlpha** [comment]
So, right now the danger area is 90 degrees left and right from some random point at the turret. As Battleshipfree already argued, it should be a much tighter cone that goes in a forward direction, starting from the muzzle. It would be a bonus if vehicles with a muzzle brake would have some overpressure going left/right of the muzzle.

No idea how it works for infantry weapons, but you can get the muzzle position for vehicles with the "gunBeg" memory point, eg:
```
private _vehicle = _this;
private _turretPath = (assignedVehicleRole (gunner _vehicle)) select 1;
private _turretConfig = [_vehicle, _turretPath] call CBA_fnc_getTurret;
private _sphere = "Sign_Sphere100cm_F" createVehicle [0,0,0];
_sphere attachTo [_vehicle, [0,0,0], getText(_turretConfig >> "gunBeg")];
```

**Comment by Coryf88** [comment]
> Regarding the tanks: Overpressure occurs all around a firing tank.
> 
> ![grafik](https://user-images.githubusercontent.com/9693842/88558779-4bf97700-d02c-11ea-819c-398703e4210e.png)

_Note:_ That image is most likely based on a figure in `TM 9-2350-264-10-2, OPERATIOR'S MANUAL FOR TANK, COMBAT, FULL-TRACKED: 120-MM GUN, M1A1`. I'm not able to use it to check what the original figure nor description was, due to it isn't publicly accessible. 😞

That image seems to have several issues:
1. Why are the yellow area and hearing protection area separate if they have the same distance?
Similar images I've found has the yellow area marked as 50m.
2. 504 meters seems quite far for requiring hearing protection and oddly specific. Why 504, instead of 500?
`MCWP 3-35.3, Military Operations on Urbanized Terrain (MOUT)` has it at 704 meters. 😕
3. Why does the yellow area stop at the front of the tank?
4. If the yellow area denotes an overpressure area, it seems that would also imply that crew shouldn't be turned out when firing the main gun. [Watch the left tank.](https://youtu.be/31kaGFETQv8?start=117)

So, it would seem that the yellow area doesn't denote an overpressure area.

Similar images I've found;
_None of the sources had any kind of description of the yellow area, other than in the figure._
![Overpressure_References](https://user-images.githubusercontent.com/204475/101614516-a808da80-39d2-11eb-8107-a52141235474.png)
